music_assistant/providers/alexa/__init__.py

```diff
@@ -0,0 +1,379 @@
+
+            # Handler that delegates to AlexaProxy's all_handler
+            async def proxy_handler(request: web.Request) -> Any:
+                response = await proxy.all_handler(request)
+                if "Successfully logged in" in getattr(response, "text", ""):
+                    # Notify the callback URL
+                    async with aiohttp.ClientSession() as session:
+                        await session.get(auth_helper.callback_url)
+                    return web.Response(
+                        text="""
+                        <html>
+                            <body>
+                                <h2>Login successful!</h2>
+                                <p>You may now close this window.</p>
+                            </body>
+                        </html>
+                        """,
+                        content_type="text/html",
+                    )
+                return response
+
+            # Register GET for the base proxy path
+            mass.webserver.register_dynamic_route(proxy_path, proxy_handler, "GET")
+            # Register POST for the specific signin helper path
+            mass.webserver.register_dynamic_route(post_path, proxy_handler, "POST")
+
+            try:
+                await auth_helper.authenticate(proxy_url, timeout=300)
+                await save_cookie(login, str(values[CONF_USERNAME]))
+            except KeyError:
+                # no URL param was found so user probably cancelled the auth
+                pass
+            except Exception as error:
+                raise LoginFailed(f"Failed to authenticate with Amazon '{error}'.")
+            finally:
+                mass.webserver.unregister_dynamic_route(proxy_path)
+                mass.webserver.unregister_dynamic_route(post_path)
+
+    return (
+        ConfigEntry(
+            key=CONF_URL,
+            type=ConfigEntryType.STRING,
+            label="URL",
+            required=True,
+            default_value="amazon.com",
+        ),
+        ConfigEntry(
+            key=CONF_USERNAME,
+            type=ConfigEntryType.STRING,
+            label="E-Mail",
+            required=True,
+        ),
+        ConfigEntry(
+            key=CONF_PASSWORD,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Password",
+            required=True,
+        ),
+        ConfigEntry(
+            key=CONF_ACTION_AUTH,
+            type=ConfigEntryType.ACTION,
+            label="Authenticate with Amazon",
+            description="Click to start the authentication process.",
+            action=CONF_ACTION_AUTH,
+            depends_on=CONF_URL,
+        ),
+    )
+
+
+async def save_cookie(login: AlexaLogin, username: str) -> None:
+    """Save the cookie file for the Alexa login."""
+    if login._session is None:
+        _LOGGER.error("AlexaLogin session is not initialized.")
+        return
+
+    cookie_dir = os.path.join(os.path.expanduser("~"), ".musicassistant", "alexa")
```

Comment:
```suggestion
    cookie_dir = os.path.join(mass.storage_path, ".alexa")
```

- use the storage_path attribute on the MusicAssistant instance to get the data storage dir
- should this be stored per username maybe to avoid conflicts when multiple instances ?